Sources/SpeziHealthCharts/MetricChart/ChartViews/HealthChart.swift

```diff
@@ -0,0 +1,60 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2024 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import SwiftUI
+import HealthKit
+
+
+public struct HealthChart: View {
+    @State private var privateRange: ChartRange
+    private var privateRangeBinding: Binding<ChartRange>?
```

Comment:
Hi @Supereg! Would you mind explaining a little more about this? When I tested the UI with this setup, I saw that when I change the date range in the lowest view in the hierarchy, this binding propagated that change up to the top level as expected, and vice versa when I change the range at the highest level. That is, the views seem to re-render correctly when I change the state? Let me know if I'm misunderstanding what you mean!

Reply:
tldr: There is a slight technical different, though, it is not 100% clear if it makes any difference. I think it shouldn't impact your development work right now as the interfaces might not be fully there where they will be. Chances are it might be refactored anyways. And if it works it works.

Long answer:

You are right, it seems that from a view-update perspective everything works fine. Either it's just because the parent view refreshes or the Bindings [transaction](https://developer.apple.com/documentation/swiftui/binding/transaction) actually changes from a `Equatable` perspective.

What I head in the back of my head: `Binding` conforms to [`DynamicProperty`](https://developer.apple.com/documentation/swiftui/dynamicproperty) (basically every SwiftUI property does to know when to prepare for the next view update). SwiftUI would then just inspect every `DynamicProperty` conforming property of the type and call `update()` accordingly. However, `Optional` does not conform to `DynamicProperty`, even if its `Wrapped` type does. I  made a small example below. If `foo` is declared as optional, `update()` is never called, if you declare it as non-optional it will be called.

There is two questions here: a) does Binding contain any important logic in its `update()` method and how would it influence behavior (e.g., properties like @Environment or @State make sure to have a consistent view on the data while `body` is getting called) and b) does SwiftUI workaround this internally and has custom handling for optional types of e.g. Binding. So honestly not sure if it actually makes a difference at all.

```swift
struct Foo: DynamicProperty {
    func update() {
        print("UPDATE")
    }
}

struct MyView: View {
    var a: Foo? = Foo()

    var body: some View {
        Text("Hello World")
    }
}
```

Reply:
After discussing with @PSchmiedmayer, we had the following thoughts:

Potential workaround that addresses the fact that `Optional` does not conform to `DynamicProperty`:

Make `Optional` conform to `DynamicProperty` when it's wrapped value conforms to `DynamicProperty`. The `update()` function will then just call the wrapped values venation if it's not `nil`, and do nothing otherwise.

Note: Apple/SwiftUI may already have native support to work around this, as we are not the first to encounter this issue.